Framework/API/inc/MantidAPI/Algorithm.h

```diff
@@ -306,6 +307,34 @@ class MANTID_API_DLL Algorithm : public IAlgorithm,
   /// Virtual method - must be overridden by concrete algorithm
   virtual void exec() = 0;
 
+  virtual void execDistributed() { exec(); }
+  virtual void execMasterOnly();
+  virtual void execNonMaster(){};
+
+  /** Get correct execution mode based on input storage modes for an MPI run.
+   *
+   * The default implementation returns ExecutionMode::Invalid. Classes
```

Comment:
So entirely up to client algorithm to set this (unless they are trivialalgorithms?)